### PR TITLE
HTTP request body passed to Gateway Knots.

### PR DESCRIPTION
**Pull request by tomaszmichalak**
---

I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: knotx-example/knotx-example-gateway/src/main/java/io/knotx/gateway/impl/RequestProcessorKnotProxyImpl.java [hunk 1]
```diff
@@ -53,19 +56,30 @@ protected KnotContext processError(KnotContext knotContext, Throwable error) {
...
+    clientResponse.setBody(Buffer.buffer(responseBody))
+        .setHeaders(getHeaders(clientResponse, responseBody));
+    clientResponse.setStatusCode(HttpResponseStatus.OK.code());
 
     return new KnotContext()
         .setClientRequest(inputContext.getClientRequest())
         .setClientResponse(clientResponse);
   }
+
+  private MultiMap getHeaders(ClientResponse clientResponse, String response) {
```
Comment (marcinczeczko) [inline knotx-example/knotx-example-gateway/src/main/java/io/knotx/gateway/impl/RequestProcessorKnotProxyImpl.java]: Looks like you don't need `response` parameter here

File: knotx-example/knotx-example-gateway/src/main/java/io/knotx/gateway/impl/RequestProcessorKnotProxyImpl.java [hunk 2]
```diff
@@ -53,19 +56,30 @@ protected KnotContext processError(KnotContext knotContext, Throwable error) {
...
+    MultiMap headers = clientResponse.getHeaders();
+    headers.add(HttpHeaders.CONTENT_LENGTH.toString().toLowerCase(),
+        Integer.toString(response.length()))
+        .add("Content-Type", "application/json");
+    return headers;
+  }
+
+  private String getResponseBodyAsString(KnotContext inputContext) {
```
Comment (marcinczeczko) [inline knotx-example/knotx-example-gateway/src/main/java/io/knotx/gateway/impl/RequestProcessorKnotProxyImpl.java]: Looks like you need this responsnse body only in line 62. If yes, return in this private method a Buffer object directly, so line 62 will looks like below
```
clientResponse.setBody(getResponseBody(inputContext))
```

File: knotx-server/src/main/java/io/knotx/server/KnotxServerVerticle.java [hunk 3]
```diff
@@ -78,13 +77,17 @@ public void start(Future<Void> fut) throws IOException, URISyntaxException {
       );
     });
 
-    if(configuration.getCustomFlow().getEngineRouting() != null) {
+    if (configuration.getCustomFlow().getEngineRouting() != null) {
       configuration.getCustomFlow().getEngineRouting().forEach((key, value) -> {
-        if (key == HttpMethod.POST) {
+        if (key != HttpMethod.GET) {
```
Comment (marcinczeczko) [inline knotx-server/src/main/java/io/knotx/server/KnotxServerVerticle.java]: What is request method is *HEAD* ?

File: knotx-example/knotx-example-gateway/src/main/java/io/knotx/gateway/impl/RequestProcessorKnotProxyImpl.java (hunk 1 above)
Comment (tomaszmichalak) [inline knotx-example/knotx-example-gateway/src/main/java/io/knotx/gateway/impl/RequestProcessorKnotProxyImpl.java]: Required to count content length.

File: knotx-server/src/main/java/io/knotx/server/KnotxServerVerticle.java (hunk 3 above)
Comment (tomaszmichalak) [inline knotx-server/src/main/java/io/knotx/server/KnotxServerVerticle.java]: Fixed. Only POST, PUT, DELETE allowed.

File: knotx-example/knotx-example-gateway/src/main/java/io/knotx/gateway/impl/RequestProcessorKnotProxyImpl.java (hunk 1 above)
Comment (marcinczeczko) [inline knotx-example/knotx-example-gateway/src/main/java/io/knotx/gateway/impl/RequestProcessorKnotProxyImpl.java]: So it's more clear if the parameter was simply a body length. currently it suggests you're doing something with the body

**Comment by tomaszmichalak** [comment]
All CR requests fixed.

**Pull request merged**